src/lib/cookies.js

```diff
@@ -64,7 +64,7 @@ const formatCookie = (cookie) => {
 class CookieHandler extends CDP {
     constructor(request) {
         super(request._client);
-        this.url = request.isNavigationRequest() ? request.url() : request.frame().url();
+        this.url = request.isNavigationRequest() ? request.url() : request.frame()?.url() || request.url();
```

Comment:
I don't think this works. Example: https://codepen.io/ardalann/pen/RwogYdj?editors=1111

Reply:
UPDATE: My example was wrong! this works: https://codepen.io/ardalann/pen/RwogYdj?editors=1111